contracts/PetPark.sol

```diff
@@ -1,7 +1,101 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.0;
 
-
 contract PetPark {
+    address public owner;
+    enum AnimalType {
+        None,
+        Fish,
+        Cat,
+        Dog,
+        Rabbit,
+        Parrot
+    }
+    enum Gender {
+        Male,
+        Female
+    }
+    enum CanBorrow {
+        Yes,
+        No
+    }
+    struct UserInfo {
+        AnimalType animal;
+        Gender gender;
+        uint256 age;
+    }
+
+    mapping(address => CanBorrow) public userCanBorrow;
+    event Added(AnimalType animal, uint256 AnimalCount);
+    event Borrowed(AnimalType animal);
+    event Returned(AnimalType animal);
+    mapping(AnimalType => uint256) public animalCounts;
+    mapping(address => uint256) public userBorrowCount;
+    mapping(address => UserInfo) public usersAllowedBorrow;
+    mapping(address => bool) public userHasBorrowed;
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Not an owner");
+        _;
+    }
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    function add(AnimalType _animal, uint256 _count) public onlyOwner {
```

Comment:
This can be a external function since it is not being used inside the contract anywhere, similarly the visibility of other functions can also be changed